Assets/LeapMotion/Core/Scripts/DataStructures/IIndexable.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using Leap.Unity.Query;
+using System.Collections.Generic;
+
+namespace Leap.Unity {
+
+  public interface IIndexable<T> {
+
+    T this[int idx] { get; }
+
+    int Count { get; }
+
+  }
+
+  public static class IIndexableExtensions {
+
+    public static IIndexableEnumerator<T> GetEnumerator<T>(this IIndexable<T> indexable) {
+      return new IIndexableEnumerator<T>(indexable);
+    }
+
+    public static QueryWrapper<T, IIndexableEnumerator<T>> Query<T>(this IIndexable<T> indexable) {
+      return new QueryWrapper<T, IIndexableEnumerator<T>>(GetEnumerator(indexable));
+    }
+
+  }
+
+  public struct IIndexableEnumerator<T> : IQueryOp<T> {
```

Comment:
If you modify the signiture for this a bit, you can have it be the enumerator for all things IIndexable, instead of needing to write a new enumerator for things like Slice and ReadonlySlice.

Something along the lines of
```
public struct IIndexableEnumerator<T, K> : IQueryOp<K> where T : IIndexable<K> {
  T indexable;
  ...
```

Reply:
Turned out to be even simpler than that! Pushed a change that just uses IIndexable's Query/Enumerator support to support Slice and ReadonlySlice

Reply:
The problem I hit with <T, K> is that that approach destroys type argument inference, which is really important for having the API be nice -- I tried it and lost the ability to just call `GetEnumerator()` and `Query()`. Turns out I was able to just use a single type argument for the element though

Reply:
New interface is gonna create garbage every time you call GetEnumerator though since it will box the slice (since slices are structs) when it references it by the IIndexable interface.